Dockerfile

```diff
@@ -1,24 +1,102 @@
-FROM ruby:2.6.3-alpine
+#### Common Build Image Stage ####
 
-ENV APP_HOME /app
+FROM ruby:2.6.3-alpine AS common-build-env
 
+ARG APP_HOME=/app
+ARG BUILD_PACKAGES="build-base"
+ARG DEV_PACKAGES="postgresql-dev git nodejs yarn"
+ARG RUBY_PACKAGES="tzdata"
+ARG bundleWithout=""
+
+ENV BUNDLER_VERSION="2.0.2"
+ENV BUNDLE_PATH="/gems"
+ENV BUNDLE_WITHOUT=${bundleWithout}
+
+RUN apk update && \
+    apk upgrade && \
+    apk add --update --no-cache $BUILD_PACKAGES $DEV_PACKAGES $RUBY_PACKAGES && \
+    gem update --system && \
+    find / -wholename '*default/bundler-*.gemspec' -delete && \
+    rm /usr/local/bin/bundle && \
+    rm /usr/local/bin/bundler && \
+    gem install bundler -v $BUNDLER_VERSION
+
+WORKDIR $APP_HOME
+    
+COPY Gemfile Gemfile.lock package.json yarn.lock ./
+
+RUN bundle install && \
+    yarn install
+
+
+#### Dev build builds on common build env image ####
+
+FROM common-build-env AS dev-build
+ARG APP_HOME=/app
+ARG railsEnv="development"
+ENV RAILS_ENV=${railsEnv}
```

Comment:
I don't understand what the ARG is for here: the stage is called `dev-build` and it's explicitly made the target in `docker-compose.override.yml`. Why not hardcode `ENV RAILS_ENV="development"`? (Also applies to the `railsEnv` in `prod-minify`)

Reply:
The ARG is the only way you can pass in variables into a docker image at build time. I can happily hardcode it at each stage if you want me to, but I felt this gave us the ability to override the default value for RAILS_ENV should the need arise in the future.
Let me know you preference and I'll make the change if required.

Reply:
I think I prefer it hardcoded as it's more to process when reading the file. Could we add it when we first need the feature?

Reply:
@duncanjbrown I have committed this change which is currently building.